### PR TITLE
Allow compilation with -DGT_ENABLE_OPENMP=OFF

### PR DESCRIPTION
**Pull request by wdeconinck**
I had a runtime error in one of my atlas unit-tests where the OpenMP runtime from GNU was conflicting with the OpenACC runtime from PGI.
In atlas I have a similar option ```-DENABLE_OMP=OFF```, but it had no effect due to GridTools which had detected OpenMP and anyway just propagated it downstream.
This option would quickly avoid any OpenMP related flags / libraries.

### CODE REVIEW

File: cmake/detect_features.cmake [hunk 1]
```diff
@@ -28,7 +28,20 @@ macro(detect_cuda)
...
+
+    set( _find_openmp FALSE )
     if (NOT DEFINED GT_ENABLE_BACKEND_X86 AND NOT DEFINED GT_ENABLE_BACKEND_MC AND NOT DEFINED GT_ENABLE_BACKEND_NAIVE)
+        set( _find_openmp TRUE )
+    elseif (GT_ENABLE_BACKEND_X86 OR GT_ENABLE_BACKEND_MC OR GT_ENABLE_BACKEND_NAIVE)
+        set( _find_openmp TRUE )
+    endif()
+
+    option( GT_ENABLE_OPENMP "Enable OpenMP backends" ${_find_openmp} )
+
```
Comment (havogt) [inline cmake/detect_features.cmake]: Let's make it an advanced option. What do you think @lukasm91 
```suggestion
    mark_as_advanced(GT_ENABLE_OPENMP)

```

**Comment by wdeconinck** [comment]
Any updates?

**Comment by havogt** [comment]
ok to test

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
@wdeconinck Can you double-check that your setup is ok (that I didn't mess-up in resolving the merge conflict).

**Comment by havogt** [comment]
tests passed (except tave which is blocked)

**Comment by havogt** [comment]
Thanks @wdeconinck.

**Pull request merged**